### PR TITLE
#3426 VS2015+RTVS on Win 10 not setting working directory of a project 

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
The problem: we use TEMP path in local and remote. This breaks workflow when markdown has R code that sources files or reads CSV - basically any operation that involves paths.

Fix: provide work folder to the publishing code. 

1. Standalone file: use TEMP as before.
2. Project: work folder path is relative to the current working directory. Handle remote mapping as well.

```
source('./script.R') # x <- c(1:10)
```

```
plot(x)
```

work correctly in local and remote as long as file is in project.

### CODE REVIEW

File: src/Common/Core/Impl/Extensions/IOExtensions.cs [hunk 1]
```diff
@@ -22,6 +23,20 @@ public static class IOExtensions {
             return path.StartsWithIgnoreCase(bp) ? path.Substring(bp.Length) : path;
         }
 
+        // TODO: merge with above
+        public static string MakeCompleteRelativePath(this string path, string basePath) {
```
Comment (MikhailArkhipov) [inline src/Common/Core/Impl/Extensions/IOExtensions.cs]: Not too happy with the name, but don't want to change behaviour of the existing `MakeRelativePath`

File: src/Host/Client/Impl/rtvs/R/util.R [hunk 2]
```diff
@@ -204,11 +204,17 @@ export_to_pdf <- function(device_id, plot_id, width, height) {
...
-    writeBin(get_blob(blob_id), rmdpath);
+    if(identical(work_folder, '')) {
+        work_folder <- tempdir();
+    } else {
+        work_folder <- normalizePath(work_folder);
+    }
+
+    input_path <- tempfile('rmd_', tmpdir = work_folder, fileext = '.rmd');
+    writeBin(get_blob(blob_id), input_path);
+    on.exit(file.remove(input_path), add = TRUE);
```
Comment (MikhailArkhipov) [inline src/Host/Client/Impl/rtvs/R/util.R]: This also fixes problem that one of the temp files was not deleted. `on.exit` without `add = TRUE` overwrites previously set actions.

**Comment by AlexanderSher** [comment]
So what was the problem with TEMP on remote - we don't have permissions for it?

**Comment by MikhailArkhipov** [comment]
`source('./script.R')` or `read.csv("..\..\foo\myfile.csv") failed because relative path was no longer valid since file was actually in TEMP and not where user expected.

**Comment by MikhailArkhipov** [comment]
See also #3404 and #3426

**Comment by MikhailArkhipov** [comment]
Ping. Need to kick signed build and publish along with #3441

**Comment by karthiknadig** [comment]
👍

**Pull request merged**